app/src/main/java/com/google/android/stardroid/control/RealClock.java

```diff
@@ -22,7 +24,6 @@
 public class RealClock implements Clock {
   @Override
   public long getTimeInMillisSinceEpoch() {
-    // TODO(johntaylor): consider using SystemClock class.
-    return System.currentTimeMillis();
+    return SystemClock.currentThreadTimeMillis();
```

Comment:
This isn't right - [the docs](https://developer.android.com/reference/android/os/SystemClock.html#currentThreadTimeMillis()) say that `SystemClock.currentThreadTimeMillis()` is:

>milliseconds running in the current thread.

This method is supposed to return wall clock time, right?

If so I don't think any methods in SystemClock are going to help - those are mainly for measuring elapsed time between two events.

Reply:
@barbeau You are right. I was thinking about the exact same thing and reason. But as I was it was particularly mentioned to use System Clock, I went ahead to implement that. I made a PR to let it be reviewed. I think I should change it to a correct procedure to finding the current time. Should I go ahead with that? @barbeau @jaydeetay